lib/dynamic_type/src/dynamic_type/dynamic_type.h

```diff
@@ -69,6 +69,23 @@ struct Containers {
       std::type_identity<MemberTypes>...,
       std::type_identity<Templates<DynamicType>>...>;
 
+#if __cplusplus < 202002L
+  template <typename DynamicType, typename... MemberTypes>
+  using TypeIdentitiesAsTupleMaybeNoContainer =
+      TypeIdentitiesAsTuple<DynamicType, MemberTypes...>;
+#else
+  // Removing containers from type list so that it does not have issue with
```

Comment:
Does the bug happen even if your implementation of `type_identity` is used?

Reply:
This has nothing to do which `type_identity`. But instead, related to this problem:

If you have
```C++
using NestedVectors = DynamicType<Containers<std::vector>>;
```
then should `NestedVectors` have `operator<`?

For this case, `NestedVectors` will say:
I am just a `std::variant<std::monostate, std::vector<NestedVectors>>`, so I will have `operator<` if and only if `std::monostate` or `std::vector<NestedVectors>` has `operator<`.

And `std::vector` will say:
My item type is `NestedVectors`, so I will have `operator<` if and only if `NestedVectors` has `operator<`.

From https://en.cppreference.com/w/cpp/utility/variant/monostate, we know that `std::monostate` does have `operator<`, so the answer to the above question should be: yes, both `NestedVectors` and `std::vector<NestedVectors>` should have `operator<`

Mathematically, this is equivalent to solve a logic equation
```
p1 = true || p2
p2 = p1
```
For humans, it is obvious that the solution of the above equation is `p1 = true, p2 = true`, however, gcc is not smart enough to figure that out. I guess what gcc is trying to do is:
```
p1 = true || p2  // Looks like I need to evaluate p2 first in order to get p1, let me evaluate p2
p2 = p1  // looks like I need to evaluate p1 first in order to get p2, let me evaluate p1
p1 = ...  // infinite recursion 
```

Reply:
Thank you for asking this question, I think I have a better solution: we can just remove SFINAE on member types because it is always true: https://github.com/NVIDIA/Fuser/pull/1056